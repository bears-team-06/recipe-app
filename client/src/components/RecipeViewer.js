import React from "react";

const RecipeViewer = ({ id, ingredients, instructions, name }) => {
  return (
    <div>
      <div className={"title"}>{name}</div>
      <div className={"instructions"}>
        <h2>Ingredients:</h2>
        <ul>{IngredientLists(ingredients)}</ul>
      </div>
      <div className={"instructions"}>
          <h2>Instructions:</h2>
        <p>{instructions}</p>
      </div>
    </div>
  );
};

const IngredientLists = ingredients => {
  if (ingredients && Array.isArray(ingredients)) {
    return ingredients.map(ingredient => <li key={ingredient.replace(" ","")}>{ingredient}</li>);
  }
};

export default RecipeViewer;
