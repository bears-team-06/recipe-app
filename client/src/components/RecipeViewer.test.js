import React from "react";
import { shallow } from "enzyme";
import RecipeViewer from "./RecipeViewer";
describe("Recipe Viewer", () => {
  it("matches snapshot", () => {
    const RecipeViewerWrapper = shallow(<RecipeViewer />);
    expect(RecipeViewerWrapper).toMatchSnapshot();
  });
});
