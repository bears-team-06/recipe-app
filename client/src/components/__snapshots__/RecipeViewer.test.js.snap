// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe Viewer matches snapshot 1`] = `
<styled.div>
  <styled.div />
  <styled.div>
    <div
      className="instructions"
    >
      <h2>
        Ingredients:
      </h2>
      <ul />
    </div>
    <div
      className="instructions"
    >
      <h2>
        Instructions:
      </h2>
      <p />
    </div>
  </styled.div>
</styled.div>
`;
